tests/e2e/deployment/deployment_test.go

```diff
@@ -76,5 +76,141 @@ var _ = Describe("Application deployment in edge_core Testing", func() {
+					Expect(err).To(BeNil())
+					break
+				}
+			}
+			Expect(len(podlist.Items)).Should(Equal(replica))
+			utils.CheckPodRunningState(ctx.Cfg.ApiServer+AppHandler, podlist)
+		})
+
+	})
+	Context("Test application deployment using Pod spec", func() {
+		BeforeEach(func() {
+		})
+		AfterEach(func() {
+			var podlist metav1.PodList
+			podlist, err := utils.GetPods(ctx.Cfg.ApiServer+AppHandler, "")
+			Expect(err).To(BeNil())
+			for _, pod := range podlist.Items {
+				_, StatusCode := utils.DeletePods(ctx.Cfg.ApiServer+AppHandler+"/"+pod.Name)
+				Expect(StatusCode).Should(Equal(http.StatusOK))
+			}
+			utils.CheckPodDeleteState(ctx.Cfg.ApiServer+AppHandler, podlist)
+			utils.PrintTestcaseNameandStatus()
+		})
+
+		It("E2E_POD_DEPLOYMENT_1: Test application deployment in edge_core", func() {
+			var podlist metav1.PodList
+			//Generate the random string and assign as a UID
+			UID = "pod-app-" + utils.GetRandomString(5)
+			IsAppDeployed := utils.HandlePod(http.MethodPost, ctx.Cfg.ApiServer+AppHandler, UID, ctx.Cfg.AppImageUrl[0])
+			Expect(IsAppDeployed).Should(BeTrue())
+			podlist, err := utils.GetPods(ctx.Cfg.ApiServer+AppHandler, "")
+			Expect(err).To(BeNil())
+			utils.CheckPodRunningState(ctx.Cfg.ApiServer+AppHandler, podlist)
+		})
+
+		It("E2E_POD_DEPLOYMENT_2: Test application deployment and delete deployment using pod spec", func() {
+			var podlist metav1.PodList
+			//Generate the random string and assign as a UID
+			UID = "pod-app-" + utils.GetRandomString(5)
+			IsAppDeployed := utils.HandlePod(http.MethodPost, ctx.Cfg.ApiServer+AppHandler, UID, ctx.Cfg.AppImageUrl[0])
+			Expect(IsAppDeployed).Should(BeTrue())
+			podlist, err := utils.GetPods(ctx.Cfg.ApiServer+AppHandler, "")
+			Expect(err).To(BeNil())
+			utils.CheckPodRunningState(ctx.Cfg.ApiServer+AppHandler, podlist)
+			for _, pod := range podlist.Items {
+				_, StatusCode := utils.DeletePods(ctx.Cfg.ApiServer+AppHandler+"/"+pod.Name)
+				Expect(StatusCode).Should(Equal(http.StatusOK))
+			}
+			utils.CheckPodDeleteState(ctx.Cfg.ApiServer+AppHandler, podlist)
+
+		})
+		It("E2E_POD_DEPLOYMENT_3: Test application deployment and delete deployment using pod spec", func() {
+			var podlist metav1.PodList
+			//Generate the random string and assign as a UID
+			UID = "pod-app-" + utils.GetRandomString(5)
+			IsAppDeployed := utils.HandlePod(http.MethodPost, ctx.Cfg.ApiServer+AppHandler, UID, ctx.Cfg.AppImageUrl[0])
+			Expect(IsAppDeployed).Should(BeTrue())
+			podlist, err := utils.GetPods(ctx.Cfg.ApiServer+AppHandler, "")
+			Expect(err).To(BeNil())
+			utils.CheckPodRunningState(ctx.Cfg.ApiServer+AppHandler, podlist)
+			for _, pod := range podlist.Items {
+				_, StatusCode := utils.DeletePods(ctx.Cfg.ApiServer+AppHandler+"/"+pod.Name)
+				Expect(StatusCode).Should(Equal(http.StatusOK))
+			}
+			utils.CheckPodDeleteState(ctx.Cfg.ApiServer+AppHandler, podlist)
+			_, StatusCode := utils.DeletePods(ctx.Cfg.ApiServer+AppHandler+"/"+UID)
```

Comment:
I think test 2 and test 3 can be combined together. Only these 2 lines are extra in test3 correct ?